primefaces/src/test/java/org/primefaces/util/FileUploadUtilsTest.java

```diff
@@ -327,6 +358,63 @@ void requireValidFilename_Windows() {
         }
     }
 
+    @Test
+    void requireValidFilename_EncodedChars() {
+        try (MockedStatic<FileUploadUtils> fileUploadUtilsMockedStatic = Mockito.mockStatic(FileUploadUtils.class, Mockito.CALLS_REAL_METHODS)) {
+            fileUploadUtilsMockedStatic.when(FileUploadUtils::isSystemWindows).thenReturn(false);
+
+            // Check each invalid character individually
+            String[] invalidCharacters = {"%20", "%21"};
+            for (String ch : invalidCharacters) {
+                String invalidFileName = "filename_with_" + ch;
+                ValidatorException ex = assertThrows(ValidatorException.class, () -> FileUploadUtils.requireValidFilename(invalidFileName));
+                assertEquals("Invalid filename: (filename_with_" + ch + ") contains invalid character: " + ch, ex.getFacesMessage().getDetail());
+            }
+        }
+    }
+
+    @Test
+    void requireValidFilename_NotNull() {
+        // Arrange
+        String invalidFileName = null;
+
+        // Act
+        ValidatorException ex = assertThrows(ValidatorException.class, () -> FileUploadUtils.requireValidFilename(invalidFileName));
+
+        // Assert
+        assertEquals("Filename cannot be empty or null", ex.getFacesMessage().getDetail());
+    }
+
+    @Test
+    void requireValidFilename_NotBlank() {
+        // Arrange
+        String invalidFileName = " ";
+
+        // Act
+        ValidatorException ex = assertThrows(ValidatorException.class, () -> FileUploadUtils.requireValidFilename(invalidFileName));
+
+        // Assert
+        assertEquals("Filename cannot be empty or null", ex.getFacesMessage().getDetail());
+    }
+
+    @Test
+    void requireValidFilename_InvalidCharacters() {
+        // Arrange
+        String invalidFileName = "file:*";
+        String os = System.getProperty("os.name").toLowerCase();
+
+        // Act
+        ValidatorException ex = assertThrows(ValidatorException.class, () -> FileUploadUtils.requireValidFilename(invalidFileName));
+
+        // Assert
+        if (os.contains("win")) {
+            assertEquals("Invalid filename: (file:*) contains invalid character: :", ex.getFacesMessage().getDetail());
+        }
+        else {
+            assertEquals("Invalid Linux filename: file:*", ex.getFacesMessage().getDetail());
+        }
+    }
```

Comment:
why not using  `Mockito#mockStatic`?

Reply:
ok basically, everywhere where `os.contains("win")` is used

Reply:
Great idea.  I will update the tests to use mock static.